content/04-postgresql/13-reading-and-querying-data/04-optimizing-postgresql.mdx

```diff
@@ -0,0 +1,663 @@
+
+Before we move on, it is important to note that while slow query logging is incredibly useful, it can potentially have an additional performance impact.  PostgreSQL must perform additional operations to time each query and to record the results to a log.  This can impact performance and fill up hard drive space unexpectedly.
+
+It may not be a good idea to log slow queries at all times.  Instead, enable the functionality when you are actively investigating an issue and disable it when you are finished.
+
+#### Logging slow queries globally
+
+With that in mind, you can configure slow query logging globally by modifying the PostgreSQL server's configuration file.  You can also modify these values interactively, but setting good defaults in the configuration will make it easier to tweak interactively later.
+
+Open PostgreSQL's configuration file.  You can find the location of the current configuration file by typing:
+
+```sql
+SHOW config_file;
+```
+```sql
+               config_file
+-----------------------------------------
+ /etc/postgresql/14/main/postgresql.conf
+(1 row)
+```
+
+Inside the file, search for the `log_min_duration_statement` setting.  If our example output value above was read from the configuration file, it will be set to `-1` to indicate that the functionality is currently disabled.  There are also a number of other related settings that you can tweak depending on your needs:
+
+```
+. . .
+# Query logging configuration
+
+#log_min_duration_statement = -1 # -1 is disabled, 0 logs all statements
+                                 # and their durations, > 0 logs only
+                                 # statements running at least this number
+                                 # of milliseconds
+
+#log_min_duration_sample = -1    # -1 is disabled, 0 logs a sample of statements
+					             # and their durations, > 0 logs only a sample of
+					             # statements running at least this number
+					             # of milliseconds;
+					             # sample fraction is determined by log_statement_sample_rate
+
+#log_statement_sample_rate = 1.0 # fraction of logged statements exceeding
+                                 # log_min_duration_sample to be logged;
+                                 # 1.0 logs all such statements, 0.0 never logs
+
+#log_transaction_sample_rate = 0.0 # fraction of transactions whose statements
+                                   # are logged regardless of their duration; 1.0 logs all
+                                   # statements from all transactions, 0.0 never logs
+. . .
+```
+
+Currently, the `log_min_duration_statement` setting is commented out with its current value set to `-1` to represent the default value.  The other settings are well-commented within the file and allow you to sample statements that are over the minimum instead of logging all of the statements.  The last setting show allows you to do sampling of statements within transactions as well.
```

Comment:
I think there is an extra word or word missing in the last sentence of this paragraph.